### ISSUE TITLE
JOBDEV DQ built from SYSNET;DQXDEV > --> why?

### ISSUE THREAD

**Issue by eswenson1** [PDP-10/its#1515]
DEVICE; JOBDEV DQ provides hostname resolution support (e.g. DNS lookups).  It is currently built from SYSNET; DQXDEV > rather than SYSNET; DQDEV >.  It seems the former is a stub that just uses the host table to resolve hostnames. The latter seems like a real DNS resolver that uses external DNS services on the Internet to resolve queries.

As it stands, COMSAT uses JOBDEV DQ and appears not to be able to resolve any addresses for which it doesn't have entries in the host table. Does the real DQDEV work? Can we use it? This ticket is here to remind one of us to explore this. I suspect @bictorv knows a lot more about this topic.... Björn?

**Comment by bictorv** [comment]
DQDEV works, at least with my minor patches (see https://its.victor.se/wiki/dqdev). COMSAT does not need the resolver, if you set up TCPGAT and DOMGAT appropriately (see https://its.victor.se/wiki/mail-setup).

NAME/Finger has been fixed (see the first link), but other programs should be adapted to use DQDEV, such as Supdup, Telnet, Peek, UP/DOWN, etc etc. 

In related news, the Chaosnet Bridge Program now supports a Chaosnet DNS contact, which forwards requests to an IP DNS server (such as 130.238.19.25), so also non-IP hosts should be able to use DNS. But DQDEV would need a small hack to make it work for ITS.

**Comment by eswenson1** [comment]
As far as I know, I have mail configured properly for inbound and outbound access.  Sending email to ejs@es-its.swenson.org from the internet works fine -- and I get it fine on ES.  However, if I attempt to send mail to eric@swenson.org from ES, :MAIL immediately tells me:

```
:mail eric@swenson.org
MAIL error - No such site known: "swenson.org"
Msg:
test message
↑C
Error; No recipients!
To:
```

So it seems to want to be able to look up the destination host in the host table. I want it to just use TCPGAT to send external email.  TCPGAT is configured to the IP address on my linux host (10.0.0.55).

**Comment by eswenson1** [comment]
I took a quick look at QMAIL, and it appears to want to look up the host name in the recipient email in the host table.  See the HANLYZ label in QMAIL 614.  This function skip-returns when the host isn't found in the host table and the caller jumps to error handling with the resulting error message mentioned above.

**Comment by bictorv** [comment]
Aargh, yes, the mail system on UP has broken/burned up, and I haven't had time to dig into fixing it. It might be related to the directory filling up. Does anyone know/remember the incantation to reset/flush the queues etc? Or if anyone wants to debug the CRASH;BURNUP files...

Note that you need to set both TCPGAT and DOMGAT, I believe. I don't think I've used QMAIL, only Emacs.

**Comment by eswenson1** [comment]
Yes. QMAIL thinks all recipient address hostnames must exist in the host table. We should fix that some day.

There are a few entry points in COMSAT for initializing the data files. If you have no data files (you deleted them first), you can run MFINIT$g to reacte them anew. Other entry points require to to run to WAKEUP, and then calling special purpose routines to rebuild the master and queue.
